### PR TITLE
Fix targetting a concourse over a proxy

### PR DESCRIPTION
**Pull request by ArthurHlt**
When we are under a proxy `fly` command line can't target to a concourse.
The `Transport` structucture wasn't getting http proxy from env var.

This is fix this issue

### CODE REVIEW

**Comment by poblin-orange** [comment]
we are really interested in this feature, to access concourse behind corporate firewall. Any reason not to merge ?

**Comment by mariash** [comment]
Looks like this PR was rebased incorrectly and got a bunch of original commits from master. It is hard to see what are the actual changes. Could you please rebase and force push this PR?

**Comment by ArthurHlt** [comment]
Hum I didn't see that my IDE rearrange code, sorry. Actually, there is not so much changes, just three lines. I will comment them.

File: rc/targets.go [hunk 1]
```diff
@@ -97,12 +97,13 @@ func NewClient(atcURL string, insecure bool) concourse.Client {
...
 
 	var transport http.RoundTripper
-
+	
 	transport = &http.Transport{
 		TLSClientConfig: tlsConfig,
 		Dial: (&net.Dialer{
 			Timeout: 10 * time.Second,
 		}).Dial,
+		Proxy: http.ProxyFromEnvironment,
```
Comment (ArthurHlt) [inline rc/targets.go]: I set the proxy according to the default transport client, see: https://golang.org/src/net/http/transport.go#L34 (line 34)

File: rc/targets.go [hunk 2]
```diff
@@ -149,6 +150,7 @@ func CommandTargetClient(selectedTarget TargetName, commandInsecure *bool) (conc
...
```
Comment (ArthurHlt) [inline rc/targets.go]: I set the proxy according to the default transport client, see: https://golang.org/src/net/http/transport.go#L34 (line 34)

File: commands/internal/hijacker/hijacker.go [hunk 3]
```diff
@@ -51,8 +51,8 @@ func (h *Hijacker) Hijack(handle string, spec atc.HijackProcessSpec, pio Process
 
 	dialer := websocket.Dialer{
 		TLSClientConfig: h.tlsConfig,
+		Proxy: http.ProxyFromEnvironment,
```
Comment (ArthurHlt) [inline commands/internal/hijacker/hijacker.go]: Add proxy to the dialer by using the http.ProxyFromEnvironment function which the function used by the default transport

**Comment by ArthurHlt** [comment]
Any news about the Pull Request ?

**Comment by o-orand** [comment]
Hi, as @poblin-orange said, it could be great to support proxy in Fly. It is consistent with new Concourse [Feature: Proxy support](concourse/concourse#319). Thanks :smile:

**Comment by vito** [comment]
@o-orand @ArthurHlt Sorry, this slipped though - we'll get to it pretty soon, thanks!

**Comment by o-orand** [comment]
@vito Great, thanks !

**Comment by ArthurHlt** [comment]
@vito thank you for the merge

**Pull request merged**